apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/DatasetsResourceTest.java

```diff
@@ -4481,6 +4487,245 @@ private DatasetItemPage assertDatasetExperimentPage(UUID datasetId, UUID experim
+                            "releases", DoubleNode.valueOf(132432432.79995))))
+                    .build();
+
+            var trace3 = factory.manufacturePojo(Trace.class)
+                    .toBuilder()
+                    .output(JsonUtils.readTree(RandomStringUtils.randomAlphanumeric(10)))
+                    .build();
+
+            var trace4 = factory.manufacturePojo(Trace.class)
+                    .toBuilder()
+                    .output(null)
+                    .build();
+
+            var trace5 = factory.manufacturePojo(Trace.class)
+                    .toBuilder()
+                    .output(JsonUtils.readTree(Set.of("test", "1233", "test2", "12338")))
+                    .build();
+
+            return List.of(trace, trace2, trace3, trace4, trace5);
+        }
+
+        @Test
+        void getExperimentItemsOutputColumns__whenFetchingByDatasetId__thenReturnColumns() {
+            var workspaceName = UUID.randomUUID().toString();
+            var apiKey = UUID.randomUUID().toString();
+            var workspaceId = UUID.randomUUID().toString();
+
+            mockTargetWorkspace(apiKey, workspaceName, workspaceId);
+
+            String projectName = RandomStringUtils.randomAlphanumeric(20);
+
+            List<Trace> traces = createTrace().stream()
+                    .map(trace -> trace.toBuilder()
+                            .projectName(projectName)
+                            .build())
+                    .toList();
+
+            traces.parallelStream().forEach(trace -> createAndAssert(trace, workspaceName, apiKey));
+
+            var dataset = factory.manufacturePojo(Dataset.class);
+            var datasetId = createAndAssert(dataset, apiKey, workspaceName);
+
+            var datasetItemBatch = factory.manufacturePojo(DatasetItemBatch.class).toBuilder()
+                    .datasetId(datasetId)
+                    .build();
+
+            putAndAssert(datasetItemBatch, workspaceName, apiKey);
+
+            // Creating 5 different experiment ids
+            var experimentIds = IntStream.range(0, 5).mapToObj(__ -> GENERATOR.generate()).toList();
+
+            List<ExperimentItem> experimentItems = IntStream.range(0, 5)
+                    .mapToObj(i -> factory.manufacturePojo(ExperimentItem.class).toBuilder()
+                            .experimentId(experimentIds.get(i))
+                            .traceId(traces.get(i).id())
+                            .datasetItemId(datasetItemBatch.items().get(i).id()).build())
+                    .toList();
+
+            var experimentItemsBatch = ExperimentItemsBatch.builder()
+                    .experimentItems(Set.copyOf(experimentItems)).build();
+
+            createAndAssert(experimentItemsBatch, apiKey, workspaceName);
+
+            Set<Column> expectedOutput = getOutputDynamicColumns(traces);
+
+            assertColumns(datasetId, apiKey, workspaceName, null, expectedOutput);
```

Comment:
for readability its nicer to have a car noExperiments = null and pass that to `assertColumns` ( vs null as arg) 

minor, test is valid